app/secrets.py

```diff
@@ -15,8 +13,52 @@
     "EQ_SERVER_SIDE_STORAGE_DATABASE_PASSWORD",
 ]
 
+EXPECTED_PRIVATE_KEYS = [
+    "EQ_SUBMISSION_SR_PRIVATE_SIGNING_KEY",
+    "EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY",
+]
+
+EXPECTED_PUBLIC_KEYS = [
+    "EQ_SUBMISSION_SDX_PUBLIC_KEY",
+    "EQ_USER_AUTHENTICATION_RRM_PUBLIC_KEY",
+]
+
 
 def validate_required_secrets(secrets):
     for required_secret in EXPECTED_SECRETS:
-        if required_secret not in secrets:
+        if required_secret not in secrets['secrets']:
             raise Exception("Missing Secret [{}]".format(required_secret))
+
+    # for required_private_key in EXPECTED_PRIVATE_KEYS:
+    #     if required_private_key not in secrets['private_keys']:
+    #         raise Exception("Missing Private Key [{}]".format(required_private_key))
+    #
+    # for required_public_key in EXPECTED_PUBLIC_KEYS:
+    #     if required_public_key not in secrets['public_keys']:
+    #         raise Exception("Missing Public Key [{}]".format(required_public_key))
+
+
+class SecretStore:
+    def __init__(self, secrets):
+        if 'secrets' in secrets:
+            self.secrets = secrets['secrets']
+        if 'private_keys' in secrets:
+            self.private_keys = secrets['private_keys']
+        if 'public_keys' in secrets:
+            self.public_keys = secrets['public_keys']
+
+    def get_secret_by_name(self, secret_name):
+        if secret_name in self.secrets:
```

Comment:
The validate_required_secrets validates that on startup so the application won't deploy if its missing a secret. If its not required then None seems like a good value to return?